lib/model/sharedpullerstate.go

```diff
@@ -29,15 +30,18 @@ type sharedPullerState struct {
 	version     protocol.Vector // The current (old) version
 
 	// Mutable, must be locked for access
-	err        error      // The first error we hit
-	fd         *os.File   // The fd of the temp file
-	copyTotal  int        // Total number of copy actions for the whole job
-	pullTotal  int        // Total number of pull actions for the whole job
-	copyOrigin int        // Number of blocks copied from the original file
-	copyNeeded int        // Number of copy actions still pending
-	pullNeeded int        // Number of block pulls still pending
-	closed     bool       // True if the file has been finalClosed.
-	mut        sync.Mutex // Protects the above
+	err              error                // The first error we hit
+	fd               *os.File             // The fd of the temp file
+	copyTotal        int                  // Total number of copy actions for the whole job
+	pullTotal        int                  // Total number of pull actions for the whole job
+	copyOrigin       int                  // Number of blocks copied from the original file
+	copyNeeded       int                  // Number of copy actions still pending
+	pullNeeded       int                  // Number of block pulls still pending
+	updated          time.Time            // Time when any of the counters above were last updated
+	closed           bool                 // True if the file has been finalClosed.
+	available        []protocol.BlockInfo // Blocks available in the temporary file
```

Comment:
As above, maybe a list of offsets referring to the blocks already listed in the `file` attribute?